go-controller/pkg/util/ndp/ra.go

```diff
@@ -0,0 +1,101 @@
+package ndp
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/mdlayher/socket"
+	"golang.org/x/sys/unix"
+)
+
+// RouterAdvertisement with mac, ips and lifetime field to send
+type RouterAdvertisement struct {
+	SourceMAC, DestinationMAC net.HardwareAddr
+	SourceIP, DestinationIP   net.IP
+	Lifetime                  uint16
+}
+
+// SendRouterAdvertisements will send variadic RAs.
```

Comment:
can you please add more than this as in it will open a new socket on the interface we pass and send the RAs :D